<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../img/favicon.ico">

	<title>Lab3 hmmer - Gen711 - Fall2016</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="../../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="../..">Gen711 - Fall2016</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../..">Home</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../AWS/">AWS</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../Syllabus/">Syllabus</a>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Lab lessons <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../Lab1_blast/">Lab1 blast</a>
</li>

                        
                            
<li class="active">
    <a href="./">Lab3 hmmer</a>
</li>

                        
                            
<li >
    <a href="../Lab4_fastq/">Lab4 fastq</a>
</li>

                        
                            
<li >
    <a href="../Lab5_mapping/">Lab5 mapping</a>
</li>

                        
                            
<li >
    <a href="../Lab5_trimming/">Lab5 trimming</a>
</li>

                        
                            
<li >
    <a href="../Lab7_transcriptome_assembly/">Lab7 transcriptome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab8_bacterial_genome_assembly/">Lab8 bacterial genome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab9_euk.genome.assembly/">Lab9 euk.genome.assembly</a>
</li>

                        
                            
<li >
    <a href="../LabX_mapping/">LabX mapping</a>
</li>

                        
                            
<li >
    <a href="../README/">README</a>
</li>

                        
                            
<li >
    <a href="../alignment/">Alignment</a>
</li>

                        
                            
<li >
    <a href="../exam_practical1/">Exam practical1</a>
</li>

                        
                            
<li >
    <a href="../khmer_diginorm/">Khmer diginorm</a>
</li>

                        
                            
<li >
    <a href="../mock.exam1/">Mock.exam1</a>
</li>

                        
                            
<li >
    <a href="../trimming/">Trimming</a>
</li>

                        
                            
<li >
    <a href="../unix/">Unix</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li >
                        <a rel="next" href="../Lab1_blast/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../Lab4_fastq/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
                
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<p>Lab3 : HMMER</p>
<p>During this lab, we will acquaint ourselves with the the software package HMMER. Your objectives are:</p>
<p>-</p>
<ol>
<li>
<p>Familiarize yourself with the software, how to execute it, how to visualize results.</p>
</li>
<li>
<p>Regarding your dataset. Characterize a few conserved domains.</p>
</li>
</ol>
<p>The HMMER manual <a href="ftp://selab.janelia.org/pub/software/hmmer3/3.1b1/Userguide.pdf">ftp://selab.janelia.org/pub/software/hmmer3/3.1b1/Userguide.pdf</a></p>
<p>The HMMER webpage: <a href="http://hmmer.janelia.org/">http://hmmer.janelia.org/</a></p>
<hr />
<blockquote>
<p>Step 1: Launch and AMI. For this exercise, we will use a c3.xlarge instance. Remember to change the permission of your key code <code>chmod 400 ~/Downloads/your.pem</code> (change your.pem to whatever you named it)</p>
</blockquote>
<pre><code>ssh -i ~/Downloads/your.pem ubuntu@ec2-???-???-???-???.compute-1.amazonaws.com
</code></pre>
<hr />
<pre><code>sudo bash
apt-get update
apt-get -y upgrade
apt-get -y install tmux git curl gcc make g++ python-dev unzip default-jre
</code></pre>
<p>-</p>
<blockquote>
<p>Ok, for this lab we are going to use HMMER</p>
</blockquote>
<p>-</p>
<pre><code>cd $HOME
wget http://selab.janelia.org/software/hmmer3/3.1b1/hmmer-3.1b1-linux-intel-x86_64.tar.gz
tar -zxf hmmer-3.1b1-linux-intel-x86_64.tar.gz
cd hmmer-3.1b1-linux-intel-x86_64/
./configure
make &amp;&amp; make all install
make check
</code></pre>
<hr />
<p>-</p>
<blockquote>
<p>You will download one of the 5 different datasets (use the same dataset). Do you remember how to use the <code>wget</code> and <code>gzip</code> commands from last week? Also, download Swissprot and Pfam-A</p>
</blockquote>
<p>-</p>
<pre><code>cd /mnt

#download your dataset

dataset1= https://www.dropbox.com/s/srfk4o2bh1qmq6l/dataset1.fa.gz
dataset2= https://www.dropbox.com/s/977n0ibznzuor22/dataset2.fa.gz
dataset3= https://www.dropbox.com/s/8s2h7sm6xtoky6q/dataset3.fa.gz
dataset4= https://www.dropbox.com/s/qth3mjrianb48a6/dataset4.fa.gz
dataset5= https://www.dropbox.com/s/quexoxfh6ttmudo/dataset5.fa.gz

#download the SwissProt database

wget ftp://ftp.uniprot.org/pub/databases/uniprot/current_release/knowledgebase/complete/uniprot_sprot.fasta.gz

#download the Pfam-A database

wget ftp://ftp.ebi.ac.uk/pub/databases/Pfam/current_release/Pfam-A.hmm.gz
</code></pre>
<blockquote>
<p>we are going to run HMMER to identify conserved protein domains. This will take a little while, and we'll use <code>tmux</code> to allow us to do this in the background, and continue to work on other things.</p>
</blockquote>
<pre><code>gzip -d *gz
tmux new -s pfam
hmmpress Pfam-A.hmm #this is analogous to 'makeblastdb'
hmmscan -E 1e-3 --domtblout dataset.pfam --cpu 4 Pfam-A.hmm dataset1.fa
ctl-b d
top -c #see that hmmscan is running..
</code></pre>
<blockquote>
<p>the neat thing about HMMER is that it can be used as a replacement for blastP or PSI-blast.</p>
</blockquote>
<pre><code>#blastp-like HBB-HUMAN is a Hemoglobin B protein sequence.

phmmer --domtblout hbb.phmmer -E 1e-5 \
/home/ubuntu/hmmer-3.1b1-linux-intel-x86_64/tutorial/HBB_HUMAN \
uniprot_sprot.fasta

#PSI-blast-like

jackhmmer --domtblout hbb.jackhammer -E 1e-5 \
/home/ubuntu/hmmer-3.1b1-linux-intel-x86_64/tutorial/HBB_HUMAN \
uniprot_sprot.fasta

#you can look at the results using `more hmm.phmmer` or `more hmm.jackhmmer`. Try blasting a few of the results using the BLAST web interface.
</code></pre>
<blockquote>
<p>Now let's look at the Pfam results. This analyses may still be running, but we can look at it while it's still in progress.</p>
</blockquote>
<pre><code>more dataset.pfam
#There are a bunch of columns in this table - what do they mean?

#Try to extract all the hits to a specific domain. Google a few domains (column 1) to see if any seem interesting.

#for instance, find all occurrences of ABC_tran
grep ABC_tran dataset.pfam

#use grep to count the number of matches. Copy this number down.

grep -c ABC_tran dataset.pfam

#Find all the contigs that have a ABC_tran domain.

grep ABC_tran dataset.pfam | awk '{print $4}' | sort | uniq
</code></pre>
<blockquote>
<p>Just for fun, check on the Pfam search to see what it is doing... </p>
</blockquote>
<pre><code>tmux attach -t pfam
ctl-b d
</code></pre></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>
