<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../img/favicon.ico">

	<title>Lab6 khmer - Gen711 - Fall2016</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="../../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="../..">Gen711 - Fall2016</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../..">Home</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../AWS/">AWS</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../Syllabus/">Syllabus</a>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Lab lessons <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../Lab10_bacterial_genome_assembly/">Lab10 bacterial genome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab1_blast/">Lab1 blast</a>
</li>

                        
                            
<li >
    <a href="../Lab3_hmmer/">Lab3 hmmer</a>
</li>

                        
                            
<li >
    <a href="../Lab4_fastq/">Lab4 fastq</a>
</li>

                        
                            
<li >
    <a href="../Lab5_mapping/">Lab5 mapping</a>
</li>

                        
                            
<li >
    <a href="../Lab5_trimming/">Lab5 trimming</a>
</li>

                        
                            
<li class="active">
    <a href="./">Lab6 khmer</a>
</li>

                        
                            
<li >
    <a href="../Lab7_transcriptome_assembly/">Lab7 transcriptome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab8_mapping/">Lab8 mapping</a>
</li>

                        
                            
<li >
    <a href="../Lab9_euk.genome.assembly/">Lab9 euk.genome.assembly</a>
</li>

                        
                            
<li >
    <a href="../README/">README</a>
</li>

                        
                            
<li >
    <a href="../alignment/">Alignment</a>
</li>

                        
                            
<li >
    <a href="../mock.exam1/">Mock.exam1</a>
</li>

                        
                            
<li >
    <a href="../trimming/">Trimming</a>
</li>

                        
                            
<li >
    <a href="../unix/">Unix</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li >
                        <a rel="next" href="../Lab5_trimming/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../Lab7_transcriptome_assembly/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
                
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#lab-6-khmer">Lab 6: khmer</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h2 id="lab-6-khmer">Lab 6: khmer</h2>
<hr />
<p>During this lab, we will acquaint ourselves with digital normalization. You will:</p>
<ol>
<li>
<p>Install software and download data</p>
</li>
<li>
<p>Quality and adapter trim data sets.</p>
</li>
<li>
<p>Apply digital normalization to the dataset.</p>
</li>
<li>
<p>Count and compare kmers and kmer distributions in the normalized and un-normalized dataset.</p>
</li>
<li>
<p>Plot in RStudio.</p>
</li>
</ol>
<p>-</p>
<p>The JellyFish manual: <a href="ftp://ftp.genome.umd.edu/pub/jellyfish/JellyfishUserGuide.pdf">ftp://ftp.genome.umd.edu/pub/jellyfish/JellyfishUserGuide.pdf</a></p>
<p>The Khmer manual: <a href="http://khmer.readthedocs.org/en/v1.1/">http://khmer.readthedocs.org/en/v1.1/</a></p>
<hr />
<blockquote>
<p>Step 1: Launch and AMI. For this exercise, we will use a c3.xlarge instance. Remember to change the permission of your key code <code>chmod 400 ~/Downloads/????.pem</code> (change ????.pem to whatever you named it)</p>
</blockquote>
<pre><code>ssh -i ~/Downloads/?????.pem ubuntu@ec2-???-???-???-???.compute-1.amazonaws.com
</code></pre>
<hr />
<blockquote>
<p>Update Software</p>
</blockquote>
<pre><code>sudo bash
apt-get update
</code></pre>
<hr />
<blockquote>
<p>Install updates</p>
</blockquote>
<pre><code>apt-get -y upgrade
</code></pre>
<hr />
<blockquote>
<p>Install other software</p>
</blockquote>
<pre><code>apt-get -y install tmux git curl gcc make g++ python-dev unzip default-jre python-pip zlib1g-dev
</code></pre>
<hr />
<blockquote>
<p>Install Trimmomatic</p>
</blockquote>
<pre><code>cd $HOME
wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.32.zip
unzip Trimmomatic-0.32.zip
cd Trimmomatic-0.32
chmod +x trimmomatic-0.32.jar
</code></pre>
<hr />
<blockquote>
<p>Install Jellyfish</p>
</blockquote>
<pre><code>cd $HOME
wget ftp://ftp.genome.umd.edu/pub/jellyfish/jellyfish-2.1.3.tar.gz
tar -zxf jellyfish-2.1.3.tar.gz
cd jellyfish-2.1.3/
./configure
make -j4
PATH=$PATH:$(pwd)/bin
</code></pre>
<hr />
<blockquote>
<p>Install Khmer</p>
</blockquote>
<pre><code>cd $HOME
pip install screed pysam
git clone https://github.com/ged-lab/khmer.git
cd khmer
make -j4
make install
PATH=$PATH:$(pwd)/scripts
</code></pre>
<hr />
<blockquote>
<p>Download data. For this lab, we'll be using a smaller dataset that consists of 10million paired end reads.</p>
</blockquote>
<p>-</p>
<pre><code>cd /mnt
wget https://s3.amazonaws.com/gen711/raw.10M.SRR797058_1.fastq.gz
wget https://s3.amazonaws.com/gen711/raw.10M.SRR797058_2.fastq.gz
</code></pre>
<hr />
<blockquote>
<p>Trim low quality bases and adapters from dataset. These files will form the basis of all out subsequent analyses.</p>
</blockquote>
<p>-</p>
<pre><code>mkdir /mnt/trimming
cd /mnt/trimming

#paste the below lines together as 1 command

java -Xmx10g -jar $HOME/Trimmomatic-0.32/trimmomatic-0.32.jar PE \
-threads 4 -baseout P2.trimmed.fastQ \
/mnt/raw.10M.SRR797058_1.fastq.gz \
/mnt/raw.10M.SRR797058_2.fastq.gz \
ILLUMINACLIP:$HOME/Trimmomatic-0.32/adapters/TruSeq3-PE.fa:2:30:10 \
SLIDINGWINDOW:4:2 \
LEADING:2 \
TRAILING:2 \
MINLEN:25
</code></pre>
<hr />
<blockquote>
<p>Run Jellyfish on the un-normalized dataset.</p>
</blockquote>
<pre><code>mkdir /mnt/jelly
cd /mnt/jelly

jellyfish count -m 25 -F2 -s 700M -t 4 -C -o trimmed.jf /mnt/trimming/P2.trimmed.fastQ_1P /mnt/trimming/P2.trimmed.fastQ_2P
jellyfish histo trimmed.jf -o trimmed.histo
</code></pre>
<hr />
<blockquote>
<p>Run Khmer</p>
</blockquote>
<pre><code>mkdir /mnt/khmer
cd /mnt/khmer
interleave-reads.py /mnt/trimming/P2.trimmed.fastQ_1P /mnt/trimming/P2.trimmed.fastQ_2P -o interleaved.fq
normalize-by-median.py -p -x 15e8 -k 25 -C 50 --out khmer_normalized.fq interleaved.fq
</code></pre>
<hr />
<blockquote>
<p>Run Khmer on the normalized dataset.</p>
</blockquote>
<pre><code>cd /mnt/jelly

jellyfish count -m 25 -s 700M -t 4 -C -o khmer.jf /mnt/khmer/khmer_normalized.fq
jellyfish histo khmer.jf -o khmer.histo
</code></pre>
<blockquote>
<p>Open up a new terminal window using the buttons command-t</p>
</blockquote>
<pre><code>scp -i ~/Downloads/????.pem ubuntu@ec2-??-???-???-??.compute-1.amazonaws.com:/mnt/jelly/*histo ~/Downloads/
</code></pre>
<blockquote>
<p>Now, on your MAC, find the files you just downloaded - for the zip files - double click and that should unzip them.. Click on the <code>html</code> file, which will open up your browser. Look at the results. Try to figure out what each plot means.</p>
</blockquote>
<p>-</p>
<blockquote>
<p>Now look at the <code>.histo</code> file, which is a kmer distribution. I want you to plot the distribution using R and RStudio.</p>
</blockquote>
<p>-</p>
<blockquote>
<p>OPEN RSTUDIO</p>
</blockquote>
<pre><code>#Import all 2 histogram datasets: this is the code for importing 1 of them..

khmer &lt;- read.table("~/Downloads/khmer.histo", quote="\"")
trim &lt;- read.table("~/Downloads/trimmed.histo", quote="\"")

#What does this plot show you??

barplot(c(trim$V2[1],khmer$V2[1]),
    names=c('Non-normalized', 'C50 Normalized'),
    main='Number of unique kmers')

# plot differences between non-unique kmers

plot(khmer$V2[10:300] - trim$V2[10:300], type='l',
    xlim=c(10,300), xaxs="i", yaxs="i", frame.plot=F,
    ylim=c(-10000,60000), col='red', xlab='kmer frequency',
    lwd=4, ylab='count',
    main='Diff in 25mer counts of \n normalized vs. un-normalized datasets')
abline(h=0)
</code></pre>
<hr />
<p><a href="http://genomebio.org/Gen711/wp-content/uploads/2014/10/khmer_norm1.jpeg"><img class="aligncenter size-large wp-image-240" src="http://genomebio.org/Gen711/wp-content/uploads/2014/10/khmer_norm1-1024x700.jpeg" alt="khmer_norm" width="1024" height="700" /></a></p>
<p>-</p>
<p>-</p>
<blockquote>
<p>What do the analyses of kmer counts tell you?</p>
</blockquote></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>
