<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../img/favicon.ico">

	<title>Lab5 trimming - Gen711 - Fall2016</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="../../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="../..">Gen711 - Fall2016</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../..">Home</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../AWS/">AWS</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../Syllabus/">Syllabus</a>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Lab lessons <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../Lab1_blast/">Lab1 blast</a>
</li>

                        
                            
<li >
    <a href="../Lab3_hmmer/">Lab3 hmmer</a>
</li>

                        
                            
<li >
    <a href="../Lab4_fastq/">Lab4 fastq</a>
</li>

                        
                            
<li >
    <a href="../Lab5_mapping/">Lab5 mapping</a>
</li>

                        
                            
<li class="active">
    <a href="./">Lab5 trimming</a>
</li>

                        
                            
<li >
    <a href="../Lab7_transcriptome_assembly/">Lab7 transcriptome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab8_bacterial_genome_assembly/">Lab8 bacterial genome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab9_euk.genome.assembly/">Lab9 euk.genome.assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab9_transcriptome/">Lab9 transcriptome</a>
</li>

                        
                            
<li >
    <a href="../LabX_mapping/">LabX mapping</a>
</li>

                        
                            
<li >
    <a href="../README/">README</a>
</li>

                        
                            
<li >
    <a href="../alignment/">Alignment</a>
</li>

                        
                            
<li >
    <a href="../exam_practical1/">Exam practical1</a>
</li>

                        
                            
<li >
    <a href="../khmer_diginorm/">Khmer diginorm</a>
</li>

                        
                            
<li >
    <a href="../mock.exam1/">Mock.exam1</a>
</li>

                        
                            
<li >
    <a href="../trimming/">Trimming</a>
</li>

                        
                            
<li >
    <a href="../unix/">Unix</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li >
                        <a rel="next" href="../Lab5_mapping/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../Lab7_transcriptome_assembly/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
                
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#lab-5-trimming">Lab 5: Trimming</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h2 id="lab-5-trimming">Lab 5: Trimming</h2>
<hr />
<p>During this lab, we will acquaint ourselves with the the software packages FastQC and JellyFish. Your objectives are:</p>
<p>-</p>
<ol>
<li>
<p>Familiarize yourself with the software, how to execute it, how to visualize results.</p>
</li>
<li>
<p>Regarding your dataset. Characterize sequence quality.</p>
</li>
</ol>
<p>The FastQC manual: <a href="http://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc">http://www.bioinformatics.babraham.ac.uk/projects/download.html#fastqc</a></p>
<p>The JellyFish manual: <a href="ftp://ftp.genome.umd.edu/pub/jellyfish/JellyfishUserGuide.pdf">ftp://ftp.genome.umd.edu/pub/jellyfish/JellyfishUserGuide.pdf</a></p>
<hr />
<blockquote>
<p>Step 1: Launch and AMI. For this exercise, we will use a c3.xlarge instance. Remember to change the permission of your key code <code>chmod 400 ~/Downloads/????.pem</code> (change ????.pem to whatever you named it)</p>
</blockquote>
<pre><code>ssh -i ~/Downloads/?????.pem ubuntu@ec2-???-???-???-???.compute-1.amazonaws.com
</code></pre>
<hr />
<blockquote>
<p>Update Software</p>
</blockquote>
<pre><code>sudo bash
apt-get update
</code></pre>
<hr />
<blockquote>
<p>Install updates</p>
</blockquote>
<pre><code>apt-get -y upgrade
</code></pre>
<hr />
<blockquote>
<p>Install other software</p>
</blockquote>
<pre><code>apt-get -y install tmux git curl gcc make g++ python-dev unzip default-jre
</code></pre>
<p>-</p>
<blockquote>
<p>Ok, for this lab we are going to use FastQC. There is a version available on apt-get, but it is an old version and we want to make sure that we have the most updated version.. <span style="color: #ff0000;"><strong>Make sure you know what each of these commands does, rather than blindly copying and pasting.. </strong></span></p>
</blockquote>
<p>-</p>
<pre><code>cd $HOME
wget http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.2.zip
unzip fastqc_v0.11.2.zip
cd FastQC/
chmod +x fastqc
PATH=$PATH:$(pwd)
</code></pre>
<hr />
<blockquote>
<p>Install Trimmomatic</p>
</blockquote>
<pre><code>cd $HOME
wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.32.zip
unzip Trimmomatic-0.32.zip
cd Trimmomatic-0.32
chmod +x trimmomatic-0.32.jar
</code></pre>
<hr />
<blockquote>
<p>Install Jellyfish</p>
</blockquote>
<pre><code>cd $HOME
wget ftp://ftp.genome.umd.edu/pub/jellyfish/jellyfish-2.1.3.tar.gz
tar -zxf jellyfish-2.1.3.tar.gz
cd jellyfish-2.1.3/
./configure
make
PATH=$PATH:$(pwd)/bin
</code></pre>
<hr />
<blockquote>
<p>Download data. For this lab, we'll be using only 1 sequencing file.</p>
</blockquote>
<p>-</p>
<pre><code>cd /mnt
wget https://s3.amazonaws.com/gen711/Pero360B.2.fastq.gz
</code></pre>
<hr />
<blockquote>
<p>Do 3 different trimming levels between 2 and 40. This one is trimming at a Phred score of 30 (BAD!!!) When you run your commands, you'll need to change the numbers in <code>LEADING:30</code> <code>TRAILING:30</code> <code>SLIDINGWINDOW:4:30</code> and <code>Pero360B.trim.Phred30.fastq</code> to whatever trimming level you want to use.</p>
</blockquote>
<pre><code>mkdir /mnt/trimming
cd /mnt/trimming

#paste the below lines together as 1 command

java -Xmx10g -jar $HOME/Trimmomatic-0.32/trimmomatic-0.32.jar SE \
-threads 4 \
../Pero360B.2.fastq.gz \
Pero360B.trim.Phred30.fastq \
ILLUMINACLIP:$HOME/Trimmomatic-0.32/adapters/TruSeq3-PE.fa:2:30:10 \
SLIDINGWINDOW:4:30 \
LEADING:30 \
TRAILING:30 \
MINLEN:25
</code></pre>
<hr />
<blockquote>
<p>After Trimmomatic is done, Run FastQC. You'll have to change the numbers to match the levels you trimmed at.</p>
</blockquote>
<pre><code>cd /mnt
fastqc -t 4 Pero360B.2.fastq.gz
fastqc -t 4 trimming/Pero360B.trim.Phred2.fastq
fastqc -t 4 trimming/Pero360B.trim.Phred15.fastq
fastqc -t 4 trimming/Pero360B.trim.Phred30.fastq
</code></pre>
<hr />
<blockquote>
<p>Run Jellyfish.</p>
</blockquote>
<pre><code>mkdir /mnt/jelly
cd /mnt/jelly

# You'll have to run these commands 4 separate times -
# once for each different trimmed dataset, and once for the raw dataset.
# Change the names of the input and output files..

jellyfish count -m 25 -s 200M -t 4 -C -o trim30.jf ../trimming/Pero360B.trim.Phred30.fastq
jellyfish histo trim30.jf -o trim30.histo
</code></pre>
<hr />
<blockquote>
<p>Open up a new terminal window using the buttons command-t</p>
</blockquote>
<pre><code>scp -i ~/Downloads/????.pem ubuntu@ec2-??-???-???-??.compute-1.amazonaws.com:/mnt/*zip ~/Downloads/
scp -i ~/Downloads/????.pem ubuntu@ec2-??-???-???-??.compute-1.amazonaws.com:/mnt/jelly/*histo ~/Downloads/
</code></pre>
<blockquote>
<p>Now, on your MAC, find the files you just downloaded - for the zip files - double click and that should unzip them.. Click on the <code>html</code> file, which will open up your browser. Look at the results. Try to figure out what each plot means.</p>
</blockquote>
<p>-</p>
<blockquote>
<p>Now look at the <code>.histo</code> file, which is a kmer distribution. I want you to plot the distribution using R and RStudio.</p>
</blockquote>
<p>-</p>
<blockquote>
<p>OPEN RSTUDIO</p>
</blockquote>
<pre><code>#Import all 3 histogram datasets: this is the code for importing 1 of them..

trim2 &lt;- read.table("~/Downloads/trim2.histo", quote="\"")

#Plot: Make sure and change the names to match what you import.
#What does this plot show you??

barplot(c(trim2$V2[1],trim15$V2[1],trim30$V2[1]),
    names=c('Phred2', 'Phred15', 'Phred30'),
    main='Number of unique kmers')

# plot differences between non-unique kmers

plot(trim2$V2[2:30] - trim30$V2[2:30], type='l',
    xlim=c(2,20), xaxs="i", yaxs="i", frame.plot=F,
    ylim=c(0,2000000), col='red', xlab='kmer frequency',
    lwd=4, ylab='count',
    main='Diff in 25mer counts of freq 2 to 20 \n Phred2 vs. Phred30')
</code></pre>
<blockquote>
<p>Look at the FastQC plots across the different trimming levels. Anything surprising?</p>
<p>What do the analyses of kmer counts tell you?</p>
</blockquote></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>
