<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../img/favicon.ico">

	<title>Trimming - Gen711 - Fall2016</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="../../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="../..">Gen711 - Fall2016</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../..">Home</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../AWS/">AWS</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../Syllabus/">Syllabus</a>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Lab lessons <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../Lab1_blast/">Lab1 blast</a>
</li>

                        
                            
<li >
    <a href="../Lab3_hmmer/">Lab3 hmmer</a>
</li>

                        
                            
<li >
    <a href="../Lab4_fastq/">Lab4 fastq</a>
</li>

                        
                            
<li >
    <a href="../Lab5_mapping/">Lab5 mapping</a>
</li>

                        
                            
<li >
    <a href="../Lab5_trimming/">Lab5 trimming</a>
</li>

                        
                            
<li >
    <a href="../Lab7_transcriptome_assembly/">Lab7 transcriptome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab8_bacterial_genome_assembly/">Lab8 bacterial genome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab9_euk.genome.assembly/">Lab9 euk.genome.assembly</a>
</li>

                        
                            
<li >
    <a href="../LabX_mapping/">LabX mapping</a>
</li>

                        
                            
<li >
    <a href="../README/">README</a>
</li>

                        
                            
<li >
    <a href="../alignment/">Alignment</a>
</li>

                        
                            
<li >
    <a href="../exam_practical1/">Exam practical1</a>
</li>

                        
                            
<li >
    <a href="../khmer_diginorm/">Khmer diginorm</a>
</li>

                        
                            
<li >
    <a href="../mock.exam1/">Mock.exam1</a>
</li>

                        
                            
<li class="active">
    <a href="./">Trimming</a>
</li>

                        
                            
<li >
    <a href="../unix/">Unix</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li >
                        <a rel="next" href="../mock.exam1/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../unix/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
                
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#lab4-trimming-fastq">Lab4: Trimming fastQ</a></li>
        
    
        <li class="main "><a href="#terminate-your-instance">Terminate your instance!!!</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h2 id="lab4-trimming-fastq">Lab4: Trimming fastQ</h2>
<p>Step 1: Launch the AMI. For this exercise, we will use a c4.2xlarge instance. <strong>IMPORTANT DETAIL!!!!!!!!!</strong> We need to have more hard drive space for this exercise. So when you select the machine type, don't click <code>Review and Launch</code> like you normally do. Instead, go to near the top of teh page and click <code>4. Add Storage</code>. You'll see a column labelled <code>Size (GiB)</code> with a 8 under that.. Change that 8 to 100. Now click <code>Review and Launch</code> like normal. You now have a computer with a hard drive of size 100Gb.</p>
<pre><code class="bash">ssh -i ~/Downloads/?????.pem ubuntu@ec2-???-???-???-???
</code></pre>

<p>Update Software</p>
<pre><code class="bash">sudo apt-get update &amp;&amp; sudo apt-get -y upgrade
</code></pre>

<p>Install other software</p>
<pre><code class="bash">sudo apt-get -y install build-essential git
</code></pre>

<p>Install LinuxBrew</p>
<pre><code class="bash">
cd
wget https://keybase.io/mpapis/key.asc
gpg --import key.asc
\curl -sSL https://get.rvm.io | bash -s stable --ruby
source /home/ubuntu/.rvm/scripts/rvm


sudo mkdir /home/linuxbrew
sudo chown $USER:$USER /home/linuxbrew
git clone https://github.com/Linuxbrew/brew.git /home/linuxbrew/.linuxbrew
echo 'export PATH=&quot;/home/linuxbrew/.linuxbrew/bin:$PATH&quot;' &gt;&gt; ~/.profile
echo 'export MANPATH=&quot;/home/linuxbrew/.linuxbrew/share/man:$MANPATH&quot;' &gt;&gt; ~/.profile
echo 'export INFOPATH=&quot;/home/linuxbrew/.linuxbrew/share/info:$INFOPATH&quot;' &gt;&gt; ~/.profile
source ~/.profile
brew tap homebrew/science
brew update
brew doctor


</code></pre>

<p>Download data, and uncompress them.. Let's put this in a tmux window so we can get to doing other things.. remember you need paste the tmux relevant commands one at a time.</p>
<pre><code class="bash">
tmux new -s download

cd $HOME &amp;&amp; mkdir reads &amp;&amp; cd reads
curl -LO https://s3.amazonaws.com/gen711/TruSeq3-PE.fa
curl -L https://s3.amazonaws.com/Mc_Transcriptome/Thomas_McBr1_R1.PF.fastq.gz &gt; kidney.1.fq.gz &amp;
curl -L https://s3.amazonaws.com/Mc_Transcriptome/Thomas_McBr1_R2.PF.fastq.gz &gt; kidney.2.fq.gz

#then this to get out of tmux

ctl-b d

</code></pre>

<p>Install Skewer, a trimming tool. Seqtk, a sequence manipulation tool, and Python..</p>
<pre><code class="bash">
brew install skewer seqtk python jellyfish

</code></pre>

<p>now go back to the download tmux window to see if//wait for the download</p>
<pre><code class="bash">tmux at -t download

#IMPORTANT - when it is done downloading, kill the window

ctl-d

</code></pre>

<p>Run khmer. Make sure to look at the manual.</p>
<pre><code class="bash">
tmux new -s kmer
mkdir $HOME/kmer_analysis &amp;&amp; cd $HOME/kmer_analysis

#IMPORTANT DETAIL: usually pasting things in 1 at a time is fine - except here... When you see ``\`` at the end of lines, this means copy the 2 (or 3 or 4) lines together.


#do trimming at P2

trim=2
seqtk mergepe ~/reads/kidney.1.fq.gz ~/reads/kidney.2.fq.gz \
| skewer --stdout -l 25 -m pe --mean-quality $trim --end-quality $trim -t 8 -x $HOME/reads/TruSeq3-PE.fa - \
| jellyfish count -s 1000000 -m 25 -t 8 -o $trim.counts.jf -C /dev/stdin

jellyfish histo $trim.counts.jf &gt; $trim.counts.histo

#do trimming at P30

trim=30
seqtk mergepe ~/reads/kidney.1.fq.gz ~/reads/kidney.2.fq.gz \
| skewer --stdout -l 25 -m pe --mean-quality $trim --end-quality $trim -t 8 -x $HOME/reads/TruSeq3-PE.fa - \
| jellyfish count -s 1000000 -m 25 -t 8 -o $trim.counts.jf -C /dev/stdin

jellyfish histo $trim.counts.jf &gt; $trim.counts.histo


 # to exit out of the tmux window, if you want to.

 ctl-b d

</code></pre>

<p>Wait for these things to be done.. Use <code>top -c</code> to do this.. Remember <code>q</code> gets you outta <code>top</code>.</p>
<p>Open up a new terminal (tab) or window using the buttons command-t. You're going to download the files you created on teh AWS machine to the MAC your using in the lab.</p>
<pre><code class="bash">scp -i ~/Downloads/????.pem ubuntu@ec2-??-???-???-??:~/kmer_analysis/*histo ~/Downloads/


</code></pre>

<p>Now look at the and <code>.histo</code> file.  which is the plot of quality containing both the mean quality as well as that for each tile. I want you to plot the distribution using R and RStudio.</p>
<p>OPEN RStudio - this should be installed on your Mac. These commands you'll type into RStudio, NOT the terminal.</p>
<pre><code class="bash">
#Import Data
p2 &lt;- read.table(&quot;~/Downloads/2.counts.histo&quot;, quote=&quot;\&quot;&quot;, comment.char=&quot;&quot;)
p30 &lt;- read.table(&quot;~/Downloads/30.counts.histo&quot;, quote=&quot;\&quot;&quot;, comment.char=&quot;&quot;)

par(mfcol=c(2,1))

plot(p2$V2[2:10] ~ 1, type='p', lwd=5,
    col='blue', frame.plot=F, xlab='25-mer frequency', ylab='kmer count',
    main='Kmer distribution in sample with different trimming thresholds')

lines(p30$V2[2:10] ~ 1, type='p', lwd=5,
    col='red')

plot(p2$V2[2:30] - p30$V2[2:30], type='p',
    xlim=c(2,20), xaxs=&quot;i&quot;, yaxs=&quot;i&quot;, frame.plot=F,
    ylim=c(0,1500000), col='red', xlab='kmer frequency',
    lwd=4, ylab='count',
    main='Diff in 25mer counts of freq 2 to 20 \n Phred2 vs. Phred30')
</code></pre>

<h2 id="terminate-your-instance">Terminate your instance!!!</h2></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>
