<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge">
        <meta name="viewport" content="width=device-width, initial-scale=1.0">
        
        
        
        <link rel="shortcut icon" href="../../img/favicon.ico">

	<title>Lab9 euk.genome.assembly - Gen711 - Fall2016</title>

        <link href="../../css/bootstrap-custom.min.css" rel="stylesheet">
        <link href="../../css/font-awesome-4.5.0.css" rel="stylesheet">
        <link href="../../css/base.css" rel="stylesheet">
        <link rel="stylesheet" href="../../css/highlight.css">

        <!-- HTML5 shim and Respond.js IE8 support of HTML5 elements and media queries -->
        <!--[if lt IE 9]>
            <script src="https://oss.maxcdn.com/libs/html5shiv/3.7.0/html5shiv.js"></script>
            <script src="https://oss.maxcdn.com/libs/respond.js/1.3.0/respond.min.js"></script>
        <![endif]-->

        
    </head>

    <body>

        <div class="navbar navbar-default navbar-fixed-top" role="navigation">
    <div class="container">

        <!-- Collapsed navigation -->
        <div class="navbar-header">
            
            <!-- Expander button -->
            <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".navbar-collapse">
                <span class="sr-only">Toggle navigation</span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
                <span class="icon-bar"></span>
            </button>
            

            <!-- Main title -->
            <a class="navbar-brand" href="../..">Gen711 - Fall2016</a>
        </div>

        <!-- Expanded navigation -->
        <div class="navbar-collapse collapse">
            
                <!-- Main navigation -->
                <ul class="nav navbar-nav">
                
                
                    <li >
                        <a href="../..">Home</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../AWS/">AWS</a>
                    </li>
                
                
                
                    <li >
                        <a href="../../Syllabus/">Syllabus</a>
                    </li>
                
                
                
                    <li class="dropdown active">
                        <a href="#" class="dropdown-toggle" data-toggle="dropdown">Lab lessons <b class="caret"></b></a>
                        <ul class="dropdown-menu">
                        
                            
<li >
    <a href="../Lab10_bacterial_genome_assembly/">Lab10 bacterial genome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab1_blast/">Lab1 blast</a>
</li>

                        
                            
<li >
    <a href="../Lab3_hmmer/">Lab3 hmmer</a>
</li>

                        
                            
<li >
    <a href="../Lab4_fastq/">Lab4 fastq</a>
</li>

                        
                            
<li >
    <a href="../Lab5_mapping/">Lab5 mapping</a>
</li>

                        
                            
<li >
    <a href="../Lab5_trimming/">Lab5 trimming</a>
</li>

                        
                            
<li >
    <a href="../Lab6_khmer/">Lab6 khmer</a>
</li>

                        
                            
<li >
    <a href="../Lab7_transcriptome_assembly/">Lab7 transcriptome assembly</a>
</li>

                        
                            
<li >
    <a href="../Lab8_mapping/">Lab8 mapping</a>
</li>

                        
                            
<li class="active">
    <a href="./">Lab9 euk.genome.assembly</a>
</li>

                        
                            
<li >
    <a href="../README/">README</a>
</li>

                        
                            
<li >
    <a href="../alignment/">Alignment</a>
</li>

                        
                            
<li >
    <a href="../exam_practical1/">Exam practical1</a>
</li>

                        
                            
<li >
    <a href="../mock.exam1/">Mock.exam1</a>
</li>

                        
                            
<li >
    <a href="../trimming/">Trimming</a>
</li>

                        
                            
<li >
    <a href="../unix/">Unix</a>
</li>

                        
                        </ul>
                    </li>
                
                
                </ul>
            

            <ul class="nav navbar-nav navbar-right">
                <li>
                    <a href="#" data-toggle="modal" data-target="#mkdocs_search_modal">
                        <i class="fa fa-search"></i> Search
                    </a>
                </li>
                
                    <li >
                        <a rel="next" href="../Lab8_mapping/">
                            <i class="fa fa-arrow-left"></i> Previous
                        </a>
                    </li>
                    <li >
                        <a rel="prev" href="../README/">
                            Next <i class="fa fa-arrow-right"></i>
                        </a>
                    </li>
                
                
            </ul>
        </div>
    </div>
</div>

        <div class="container">
            
                <div class="col-md-3"><div class="bs-sidebar hidden-print affix well" role="complementary">
    <ul class="nav bs-sidenav">
    
        <li class="main active"><a href="#lab-9-genome-assembly">Lab 9: Genome assembly</a></li>
        
    
    </ul>
</div></div>
                <div class="col-md-9" role="main">

<h2 id="lab-9-genome-assembly">Lab 9: Genome assembly</h2>
<hr />
<p>During this lab, we will acquaint ourselves with Genome Assembly using SPAdes. We will assembly the genome of Plasmodium falciparum. The data are taken from this paper: http://www.nature.com/ncomms/2014/140909/ncomms5754/full/ncomms5754.html?WT.ec_id=JA-NCOMMS-20140919.
As it stands right now, I think that you will do all the preprocessing steps this week, then the assembly next. Once you have done all the steps, <code>gzip</code> compress the files and download them to your USB drive, or the MAC HD. I can provide you with these files next week if issues arise. </p>
<ol>
<li>
<p>Install software and download data</p>
</li>
<li>
<p>Error correct, quality and adapter trim data sets.</p>
</li>
<li>
<p>(next week) Assemble</p>
</li>
</ol>
<p>-</p>
<p>The SPAdes manuscript: http://www.ncbi.nlm.nih.gov/pubmed/22506599
The SPAdes manual: http://spades.bioinf.spbau.ru/release3.1.1/manual.html
SPAdes website: http://bioinf.spbau.ru/spades</p>
<blockquote>
<p>Step 1: Launch and AMI. For this exercise, we will use a <span style="color: #ff0000;"><strong>c3.8xlarge</strong></span> (note different instance type). Remember to change the permission of your key code <code>chmod 400 ~/Downloads/????.pem</code> (change ????.pem to whatever you named it)</p>
</blockquote>
<pre><code>ssh -i ~/Downloads/?????.pem ubuntu@ec2-???-???-???-???.compute-1.amazonaws.com
</code></pre>
<hr />
<blockquote>
<p>Update Software</p>
</blockquote>
<pre><code>sudo bash
apt-get update
</code></pre>
<hr />
<blockquote>
<p>Install updates</p>
</blockquote>
<pre><code>apt-get -y upgrade
</code></pre>
<hr />
<blockquote>
<p>Install other software</p>
</blockquote>
<pre><code>apt-get -y install subversion tmux git curl bowtie libncurses5-dev samtools gcc make g++ python-dev unzip dh-autoreconf default-jre python-pip zlib1g-dev
</code></pre>
<hr />
<blockquote>
<p>Install Lighter, software for error correction.</p>
</blockquote>
<pre><code>cd $HOME
git clone https://github.com/mourisl/Lighter.git
cd Lighter
make -j8
PATH=$PATH:$(pwd)
</code></pre>
<hr />
<blockquote>
<p>Install Trimmomatic</p>
</blockquote>
<pre><code>cd $HOME
wget http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-0.32.zip
unzip Trimmomatic-0.32.zip
cd Trimmomatic-0.32
chmod +x trimmomatic-0.32.jar
</code></pre>
<hr />
<blockquote>
<p>Install SRAtoolkit</p>
</blockquote>
<pre><code>cd $HOME
wget http://ftp-trace.ncbi.nlm.nih.gov/sra/sdk/2.4.2/sratoolkit.2.4.2-ubuntu64.tar.gz
tar -zxf sratoolkit.2.4.2-ubuntu64.tar.gz
PATH=$PATH:/home/ubuntu/sratoolkit.2.4.2-ubuntu64/bin
</code></pre>
<hr />
<blockquote>
<p>Install SPAdes</p>
</blockquote>
<pre><code>wget http://spades.bioinf.spbau.ru/release3.1.1/SPAdes-3.1.1-Linux.tar.gz
tar -zxf SPAdes-3.1.1-Linux.tar.gz
cd SPAdes-3.1.1-Linux
PATH=$PATH:$(pwd)/bin
</code></pre>
<hr />
<blockquote>
<p>Download 3.5kb MP library</p>
</blockquote>
<pre><code>cd /mnt
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/ERR/ERR022/ERR022558/ERR022558.sra
</code></pre>
<blockquote>
<p>Download 10kb MP library</p>
</blockquote>
<pre><code>cd /mnt
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/ERR/ERR022/ERR022557/ERR022557.sra
</code></pre>
<blockquote>
<p>Download PE library #1</p>
</blockquote>
<pre><code>cd /mnt
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/ERR/ERR019/ERR019273/ERR019273.sra
</code></pre>
<hr />
<blockquote>
<p>Download PE library #2</p>
</blockquote>
<pre><code>cd /mnt
wget ftp://ftp-trace.ncbi.nlm.nih.gov/sra/sra-instant/reads/ByRun/sra/ERR/ERR019/ERR019275/ERR019275.sra
</code></pre>
<hr />
<blockquote>
<p>Extract fastQ from sra format.</p>
</blockquote>
<pre><code>cd /mnt

#this is a basic for loop. Copy is all as 1 line.

for i in `ls *sra`; do
    fastq-dump --split-files --split-spot $i;
    rm $i;
done
</code></pre>
<hr />
<blockquote>
<p>Error Correct Data</p>
</blockquote>
<pre><code>mkdir /mnt/ec
cd /mnt/ec
lighter -r /mnt/ERR019273_1.fastq -r /mnt/ERR019273_2.fastq -t 32 -k 21 45000000 .1
lighter -r /mnt/ERR022557_1.fastq -r /mnt/ERR022557_2.fastq -t 32 -k 21 45000000 .1
lighter -r /mnt/ERR022558_1.fastq -r /mnt/ERR022558_2.fastq -t 32 -k 21 45000000 .1
lighter -r /mnt/ERR019275_1.fastq -r /mnt/ERR019275_2.fastq -t 32 -k 21 45000000 .1

#remove the raw files.

rm *fastq &amp;
</code></pre>
<blockquote>
<p>trim the data:</p>
</blockquote>
<pre><code>mkdir /mnt/trim
cd /mnt/trim
#paste the below lines together as 1 command

java -Xmx10g -jar $HOME/Trimmomatic-0.32/trimmomatic-0.32.jar PE \
-threads 32 -baseout PE_lib1.fq \
/mnt/ec/ERR019273_1.cor.fq \
/mnt/ec/ERR019273_2.cor.fq \
ILLUMINACLIP:$HOME/Trimmomatic-0.32/adapters/TruSeq2-PE.fa:2:30:10 \
SLIDINGWINDOW:4:2 \
LEADING:2 \
TRAILING:2 \
MINLEN:25

#paste the below lines together as 1 command

java -Xmx10g -jar $HOME/Trimmomatic-0.32/trimmomatic-0.32.jar PE \
-threads 32 -baseout PE_lib2.fq \
/mnt/ec/ERR019275_1.cor.fq \
/mnt/ec/ERR019275_2.cor.fq \
ILLUMINACLIP:$HOME/Trimmomatic-0.32/adapters/TruSeq2-PE.fa:2:30:10 \
SLIDINGWINDOW:4:2 \
LEADING:2 \
TRAILING:2 \
MINLEN:25

#paste the below lines together as 1 command

java -Xmx10g -jar $HOME/Trimmomatic-0.32/trimmomatic-0.32.jar PE \
-threads 32 -baseout MP10000.fq \
/mnt/ec/ERR022557_1.cor.fq \
/mnt/ec/ERR022557_2.cor.fq \
ILLUMINACLIP:$HOME/Trimmomatic-0.32/adapters/TruSeq2-PE.fa:2:30:10 \
SLIDINGWINDOW:4:2 \
LEADING:2 \
TRAILING:2 \
MINLEN:25

#paste the below lines together as 1 command

java -Xmx10g -jar $HOME/Trimmomatic-0.32/trimmomatic-0.32.jar PE \
-threads 32 -baseout MP3500.fq \
/mnt/ec/ERR022558_1.cor.fq \
/mnt/ec/ERR022558_2.cor.fq \
ILLUMINACLIP:$HOME/Trimmomatic-0.32/adapters/TruSeq2-PE.fa:2:30:10 \
SLIDINGWINDOW:4:2 \
LEADING:2 \
TRAILING:2 \
MINLEN:25
</code></pre>
<hr />
<pre><code>mkdir /mnt/

#remove the corrected files.

rm *fq
</code></pre>
<blockquote>
<p>Assembly. This you may want to do next week. Alternatively, you can put it in a tmux window and let it run. You'd have to login later however to download the assembled genome.  </p>
</blockquote>
<pre><code>mkdir spades
cd spades

spades.py -t 32 -m 60 \
--pe1-1 /mnt/trim/PE_lib1_1P.fq \
--pe1-2 /mnt/trim/PE_lib1_2P.fq \
--pe2-1 /mnt/trim/PE_lib2_1P.fq \
--pe2-2 /mnt/trim/PE_lib2_2P.fq \
--mp1-1 /mnt/trim/MP3500_1P.fq \
--mp1-2 /mnt/trim/MP3500_2P.fq \
--mp2-1 /mnt/trim/MP10000_1P.fq \
--mp2-2 /mnt/trim/MP10000_2P.fq \
-o Pfal --only-assembler
</code></pre></div>
            
        </div>

        <footer class="col-md-12">
            <hr>
            
            <p>Documentation built with <a href="http://www.mkdocs.org/">MkDocs</a>.</p>
        </footer>

        <script src="../../js/jquery-1.10.2.min.js"></script>
        <script src="../../js/bootstrap-3.0.3.min.js"></script>
        <script src="../../js/highlight.pack.js"></script>
        <script>var base_url = '../..';</script>
        <script data-main="../../mkdocs/js/search.js" src="../../mkdocs/js/require.js"></script>
        <script src="../../js/base.js"></script>

        <div class="modal" id="mkdocs_search_modal" tabindex="-1" role="dialog" aria-labelledby="Search Modal" aria-hidden="true">
            <div class="modal-dialog">
                <div class="modal-content">
                    <div class="modal-header">
                        <button type="button" class="close" data-dismiss="modal"><span aria-hidden="true">&times;</span><span class="sr-only">Close</span></button>
                        <h4 class="modal-title" id="exampleModalLabel">Search</h4>
                    </div>
                    <div class="modal-body">
                        <p>
                            From here you can search these documents. Enter
                            your search terms below.
                        </p>
                        <form role="form">
                            <div class="form-group">
                                <input type="text" class="form-control" placeholder="Search..." id="mkdocs-search-query">
                            </div>
                        </form>
                        <div id="mkdocs-search-results"></div>
                    </div>
                    <div class="modal-footer">
                    </div>
                </div>
            </div>
        </div>

    </body>
</html>
